### ISSUE TITLE
[BUG] -AccessToken parameter missing from Connect-PnPOnline cmdlet

### ISSUE THREAD

**Issue by Semptra** [pnp/powershell#221]
### Reporting an Issue or Missing Feature
When trying to use 1.2.0 version of PnP (https://www.powershellgallery.com/packages/PnP.PowerShell/1.2.0) I've noticed that there are no `-AccessToken` parameter for `Connect-PnPOnline` cmdlet. It seems like it's been disabled in this commit (https://github.com/pnp/powershell/commit/75e422641f234817f6ccfadd7ddb0e65af812b24). Is there a reason for this? Our application could only use access token in order to connect, so having this parameter is critical. Will this `-AccessToken` parameter be available in the later versions?

### Expected behavior 
Be able to connect using the ` Connect-PnPOnline -Url $Url -AccessToken $AccessToken`

### Actual behavior 
Not able to connect because `-AccessToken` parameter is missing.

### Steps to reproduce behavior
```powershell
PS > Connect-PnPOnline -Url $Url -AccessToken $AccessToken
Connect-PnPOnline: A parameter cannot be found that matches parameter name 'AccessToken'.
```

### What is the version of the Cmdlet module you are running?
1.2.0

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by erwinvanhunen** [comment]
Hi, that is correct. We are following guideliness from the MSAL team here which states that access tokens should not be handled by users of products. E.g. tools can use access tokens internally, but those should not be exposed as such.

We moved over to MSAL behind the scenes and we handle token acquisition and renewal fully automated based upon the MSAL library. Providing an access token manually would break that as we do not have access to the refreshtoken which becomes an issue when the token expires. The result is that a script can fail the moment the token expired. Also, we check the acquired token for the appropriate audience if the token does not contain the correct audience we try, using MSAL, to acquire a new token with those permissions.

Can you describe the scenario why you need to provide an external token to connect?

**Comment by Semptra** [comment]
Our application uses the following flow:
1. User connects using the Microsoft OAuth 2.0 authorization code flow (https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow). This flow returns an access and refresh tokens. Our app persists access token and refresh token, so the user has to follow authorization flow only once. After that we are responsible for further token exchange flows.
2. Then we connect to SharePointOnlinePnP using the retrieved access token using the `Connect-PnPOnline -Url $TenantUrl -AccessToken $AccessToken` cmdlet.
3. If the access token expires, we refresh it internally using the same Microsoft Identity flow and then connect to SPO PnP again with the new access token.

**Comment by erwinvanhunen** [comment]
Are you using MSAL in your app? And if so, are you using the standard token caching functionality used there? If so it could be solved, although it (right now) would require a fork of PnP PowerShell that you modify slighty and compile yourself. We use the AuthenticationManager from the PnP Framework in PnP PowerShell to acquire tokens. The AuthenticationManager uses MSAL to acquire tokens and keep them valid. We expose the option to provide an action (Action<ITokenCache>) that will be called if specified. Using the parameter sent to the action you can register your own cache.

By default, PnP PowerShell uses the in memory instance of the cache, but if at initialization time of the authentication manager you register your own cache, which is the same then as the one used in your app you will be able to share tokens between your app and pnp powershell. Result is that if you authenticate with the same credentials you will not do a round trip to the Azure AD but PnP PowerShell will acquire the token from the cache.

**Comment by erwinvanhunen** [comment]
after discussing this in the team we decided to return the -AccessToken parameter. Keep in mind that functionality will be limited and you are fully in charge yourself to make sure that the access token is valid and issued for the correct audience. This functionality will be available as of the 1.2.19-nightly.
